src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -90,6 +102,132 @@ public static OAuthAuthResponse getAuthorizationUrl(Main main, AppIdentifier app
         return new OAuthAuthResponse(redirectTo, cookies);
     }
 
+    //This more or less acts as a pass-through for the sdks, apart from camelCase <-> snake_case key transformation and setting a few default values
+    public static JsonObject registerOAuthClient(Main main, AppIdentifier appIdentifier, Storage storage, JsonObject paramsFromSdk)
+            throws TenantOrAppNotFoundException, InvalidConfigException, IOException, OAuthClientRegisterException,
+            NoSuchAlgorithmException, StorageQueryException {
+
+        OAuthStorage oauthStorage = StorageUtils.getOAuthStorage(storage);
+        String adminOAuthProviderServiceUrl = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main).getOAuthProviderAdminServiceUrl();
+
+        byte[] idBaseBytes = new byte[48];
+
+        while(true){
+            new SecureRandom().nextBytes(idBaseBytes);
+            String clientId = "supertokens_" + Utils.hashSHA256Base64UrlSafe(idBaseBytes);
+            try {
+                if(oauthStorage.isClientIdAlreadyExists(clientId)){
+                    continue; // restart
+                }
+
+                JsonObject hydraRequestBody = constructHydraRequestParamsForRegisterClientPOST(paramsFromSdk, clientId);
+                JsonObject hydraResponse = HttpRequest.sendJsonPOSTRequest(main, "", adminOAuthProviderServiceUrl + HYDRA_CLIENTS_ENDPOINT, hydraRequestBody, 10000, 10000, null);
+
+                oauthStorage.addClientForApp(appIdentifier, clientId);
+
+                return formatResponseForSDK(hydraResponse); //sdk expects everything from hydra in camelCase
+            } catch (HttpResponseException e) {
+                try {
+                    throw createCustomExceptionFromHttpResponseException(e, OAuthClientRegisterException.class);
+                } catch (NoSuchMethodException | InvocationTargetException | InstantiationException |
+                         IllegalAccessException ex) {
+                    throw new RuntimeException(ex);
+                }
+            } catch (OAuth2ClientAlreadyExistsForAppException e) {
+                //in theory, this is unreachable. We are registering new clients here, so this should not happen.
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public static JsonObject loadOAuthClient(Main main, AppIdentifier appIdentifier, Storage storage, String clientId)
+            throws TenantOrAppNotFoundException, OAuthClientException, InvalidConfigException, StorageQueryException,
+            IOException {
+        OAuthStorage oauthStorage = StorageUtils.getOAuthStorage(storage);
+
+        String adminOAuthProviderServiceUrl = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main).getOAuthProviderAdminServiceUrl();
+
+        if (!oauthStorage.doesClientIdExistForThisApp(appIdentifier, clientId)) {
+            throw new OAuthClientException("Unable to locate the resource", "");
```

Comment:
```suggestion
            throw new OAuthClientIdDoesNotExistForAppException();
```

Reply:
Do we really need to differentiate the "not found for this app" from the "not found at all"? Currently, I can't see how that would be a help for us, the result in either case would be a "Couldn't find" (Does it really matter if the clientId exists for an other app, but not for the current one?) 
I renamed the `OAuthClientException` to `OAuthClientNotFoundException` so that way the same type of exception can be used to communicate this and the case where the clientId is not found in the oauth provider's database.